src/main/java/com/caiporalang/lexic/state/RealNumberState.java

```diff
@@ -0,0 +1,26 @@
+package com.caiporalang.lexic.state;
+
+import com.caiporalang.lexic.ScannerContext;
+import com.caiporalang.lexic.Token;
+import com.caiporalang.lexic.TokenType;
+import com.caiporalang.lexic.exception.MalformedNumber;
+
+public class RealNumberState extends ScannerState {
+    RealNumberState(ScannerContext context, String term) {
+        super(context, term);
+    }
+
+    @Override
+    public Token getTokenFromState(char current) {
+        if (context.isDigit(current)) {
```

Comment:
It's a bit weird to have to ask everything about the current character to the context. Context here doesn't seem to be the right class to do character validations. In a next PR I suggest moving it to another class maybe or come up with a better design (it's probably that just creating a new class just introduces a procedural/util class).